apps/extensions_manager/lib/extensions_manager/model_extension.ex

```diff
@@ -0,0 +1,75 @@
+defmodule ExtensionsManager.ModelExtension do
+  defmacro __using__(_opts) do
+    quote do
+      Module.register_attribute(__MODULE__, :schema_changes, accumulate: true)
+      Module.register_attribute(__MODULE__, :method_block, accumulate: true)
+
+      import ExtensionsManager.ModelExtension, only: [add_to_schema: 4, add_to_schema: 3, include_method: 1]
+      @before_compile ExtensionsManager.ModelExtension
+
+      defmacro __using__(_opts) do
```

Comment:
I'm thinking maybe you're attempt to implicitly change the schema definition should take a more explicit approach. Create a macro instead that adds the fields you want and just insert that macro where it needs to go.